### ISSUE TITLE
windows_certificate remotely over winrm does not work

### ISSUE THREAD

**Issue by tophercullen** [opscode-cookbooks/windows#227]
As the title says, using the windows_certificate resource over remotely over winrm doesn't work. Running locally on the machine succeeds. Remote execution fails with "Begin output of "powershell.exe" -NoLogo -NonInteractive -NoProfile -ExecutionPolicy Unrestricted -InputFormat None -File "C:/Users/ADMINI~1/AppData/Local/Temp/chef-script20150609-2888-i668tc.ps1"
STDERR: Exception calling ".ctor" with "3" argument(s): "Access denied."

Full output: http://pastebin.com/R3T4wkSb

I copied the script to the host and ran it remotely, with the same error: http://pastebin.com/cjsGXGW0

Now, this is where things get weird. If I run that exact script locally (on the remote machine), not only does  it work,  but then it also starts working remotely. Even if after removing the cert, both remote and local continue function and chef runs complete without error. The only way I can reproduce this is with a fresh host. I had my own implementation of this resource working fine. So its something specific to how this is written.

Note: this is not the first time I've run across windows treating the same code very differently local vs remote: https://github.com/opscode-cookbooks/windows/issues/196

**Comment by jdmundrawala** [comment]
Probably related to the user profile not being loaded:
Would probably be fixed by calling https://msdn.microsoft.com/en-us/library/windows/desktop/bb762281%28v=vs.85%29.aspx or by running powershell.exe through Start-Process with -LoadUserProfile

**Comment by tophercullen** [comment]
I tried Start-Proccess with -LoadUserProfile and it gave the same result. Not loading the user profile made it fail silently. However, I did get it working. The problem appears to be related to the X509KeyStorageFlags: https://msdn.microsoft.com/en-us/library/system.security.cryptography.x509certificates.x509keystorageflags(v=vs.110).aspx
I'm importing a PFX so the PresistKeySet is flag is being added. I changed, and then later added, this to the "MachineKeySet" flag and it worked.  Even after reading the docs, I still don't really understand what PersistKeySet does, or what the default behavior is for the flags. Based on the description of the UserKeySet flag, it seems the MachineKeySet would be redundant. However, for some magically reason it works and the certs end up where I expected them to.
